src/Integrated.php

```diff
@@ -2,10 +2,21 @@
 
 namespace LCache;
 
-final class Integrated
```

Comment:
many of the Integration level tests were implemented by knowing the inner workings of L1 and L2 and tweaking the inner objects directly, ruining the encapsulation idea of the class and also consistent factory interface implementation for all the tests.

By removing the **final** here I did a descendant (IntegratedMock) that added many of the needed accessors to port the tests as close as possible.